doc/README.twom.md

```diff
@@ -0,0 +1,189 @@
+- The write amplification issue is still the same, twom is basically the same format as twoskip, just with the addition of ancestor pointers and richer tombstones.
+
+### Navigating the code - data structures:
+
+Structures named `twom_` are public, `tm_` are entirely internal.
+
+- `struct twom_db` - this is refcounted and contains pointers to all the refcounted resources that it has open.  When you open a DB again, you get another pointer to the same file, and when you open a transaction or a location on the database, you hold a reference that the file won’t be closed under you.
+- `struct twom_txn` - a transaction on the database.
+    - If this is an exclusive transaction (not TWOM_SHARED) then you can write to the file and either abort or commit.  A transaction dirties the file on first write.  An exclusive transaction forces the file to remain locked exclusively for the duration of the transaction.
+    - If this is read-only transaction, it can be either MVCC or not.  If it is MVCC, then it never changes file, it always reads from the same file, though it may still need to refresh the length if the lock has been released.  Read-only transactions can release the lock, either explicitly using `twom_yield` or implicitly every N transactions to avoid starving other database users.
+    - Has a reference counted on the file into which it points, to ensure it isn’t released if this is an MVCC transaction.
+- `struct tm_loc` - a location in the database.  This contains an offset for the current record, the locations of the immediately preceeding record at every level, and the length of the file when it was calculated (a change in file length means that back pointers may have been changed and the location is no longer valid).  If the current record has a DELETE before it, will also contain a `deleted_offset` value.
+    - Has a reference counted on the file into which it points, which is required to allow relocation without having to copy the key content.
+- `struct tm_file` an mmaped file.  Has the file handle and tracks the lock on it.  Also has a parsed version of the header of the file, which is updated every time it’s locked.  (strictly this isn’t necessary, we could do it all with MMAP reads directly from the file).
+    - Also contains pointers to the checksum and comparison functions for this file.  These CANNOT BE CHANGED after file creation, but can either use the ones built into twom, or you can pass external functions at file creation time.
+    - Refcounted by txn and loc above.
+- `struct twom_cursor` a wrapper around a loc and a txn, used by public API for cursor operations.
+
+## Locking
+
+Twom uses a two-phase locking system, as described in [a stackoverflow post](https://stackoverflow.com/questions/27625597/how-to-implement-a-writer-preferring-read-write-lock-for-nix-processes).
```

Comment:
Maybe more clearer e.g.: "Twom uses fcntl(2) for locking the database file, specifically a two-phase locking system..."
Maybe of interest is https://arxiv.org/html/2403.04582v3#S4 section 4.7 describing other implementations.